traits/tests/test_regression.py

```diff
@@ -259,6 +260,25 @@ def test_on_trait_change_with_list_of_extended_names(self):
         dummy.x = 11
         self.assertTrue(model.changed)
 
+    def test_set_disallowed_exception(self):
+        class StrictDummy(HasStrictTraits):
+            foo = Int
+
+        # Regression test for enthought/traits#415
+        with self.assertRaises(TraitError):
+            StrictDummy(forbidden=53)
+
+        with self.assertRaises(TraitError):
+            StrictDummy(**{'forbidden': 53})
```

Comment:
Do we want to explicitly use a `b''` string here? In case someone adds `from __future__ import unicode_literals` at the top down the line? Maybe guarded by `if six.PY2:`

Reply:
Done. Thanks!